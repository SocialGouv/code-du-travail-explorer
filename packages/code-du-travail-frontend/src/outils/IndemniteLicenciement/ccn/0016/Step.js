import React from "react";
import { Categorie, CADRE } from "./Categorie";
import { CategoriePeriod } from "./CategoriePeriod";
import { YesNoQuestion } from "../../../common/YesNoQuestion";
import { TextQuestion } from "../../../common/TextQuestion";

function Step({ form }) {
  const state = form.getState();
  const { anciennete = 0, age = 0, categorie = "" } = state.values;
  const askDetailedCategories = anciennete >= 3 && categorie === CADRE;

  return (
    <>
      <Categorie name="categorie" />
      {askDetailedCategories && <CategoriePeriod />}
      <TextQuestion
        inputType="number"
        name="age"
        size="5"
        label="Quel etait votre âge à la date du licenciement&nbsp;?"
      />
      {age > 60 && (
        <YesNoQuestion
          label="Le salarié pourrait-il bénéficier de la retraite&nbsp;?"
          name="hasRetirementAge"
        />
      )}
    </>
  );
}

export { Step };
