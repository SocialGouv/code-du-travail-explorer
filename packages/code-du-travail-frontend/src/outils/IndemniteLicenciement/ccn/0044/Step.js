import { Title } from "@socialgouv/cdtn-react-ui";
import React from "react";
import { Field } from "react-final-form";

import { TextQuestion } from "../../../common/TextQuestion";
import { YesNoQuestion } from "../../../common/YesNoQuestion";
import { EchelonChimie } from "./EchelonChimie";

function Step() {
  return (
    <>
      <Title as="h3">Chimie</Title>
      <EchelonChimie name="groupe" />

      <Field name="anciennete">
        {({ input }) => {
          const anciennete = Math.floor(input.value);
          if ((anciennete >= 1 && anciennete < 2) || anciennete > 5) {
            return (
              <>
                <YesNoQuestion
                  name="hasOpe"
                  label="Est-ce que l'entreprise est affiliée à une des organisations patronales employeur&nbsp;?"
                />
                <YesNoQuestion
                  name="isEco"
                  label="Le motif du licenciement est-il économique ?"
                />
                {anciennete > 5 && (
                  <TextQuestion
                    inputType="number"
                    name="age"
                    label="Quel était l’âge du salarié à la date du licenciement ?"
                  />
                )}
              </>
            );
          }
          return null;
        }}
      </Field>
    </>
  );
}

export { Step };
