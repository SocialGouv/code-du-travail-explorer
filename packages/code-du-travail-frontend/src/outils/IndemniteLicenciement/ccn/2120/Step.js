import React from "react";
import { Field } from "react-final-form";
import styled from "styled-components";
import { theme } from "@socialgouv/react-ui";

import { SectionTitle, Label } from "../../../common/stepStyles";

export const DISCIPLINAIRE = "disciplinaire";
export const NON_DISCIPLINAIRE = "non-disciplinaire";
export const ECONOMIQUE = "eco";

export const CADRE = "cadre";
export const NON_CADRE = "non-cadre";
export const NE_SAIT_PAS = "ne-sais-pas";

export const optionMotifs = {
  [DISCIPLINAIRE]: "Motif disciplinaire",
  [NON_DISCIPLINAIRE]: "Motif non disciplinaire",
  [ECONOMIQUE]: "Motif économique"
};

export const optionCategorie = {
  [NON_CADRE]: "Non cadre",
  [CADRE]: "Cadre",
  [NE_SAIT_PAS]: "Ne sait pas"
};

function Step() {
  return (
    <>
      <SectionTitle>Motif du licenciement</SectionTitle>
      <Field name="motif">
        {({ input }) => {
          return (
            <FormGroup>
              <Label htmlFor="input-motif">
                Quel est le motif du licenciement&nbsp;?
              </Label>
              <Select id="input-motif" {...input}>
                <option value="" disabled>
                  motifs...
                </option>
                {Object.entries(optionMotifs).map(([value, label]) => (
                  <option key={value} value={value}>
                    {label}
                  </option>
                ))}
              </Select>
            </FormGroup>
          );
        }}
      </Field>

      <Field name="categorie">
        {({ input }) => {
          return (
            <FormGroup>
              <Label htmlFor="input-categorie">
                Quelle est la catégorie professionnelle du salarié&nbsp;?
              </Label>

              <Select id="input-categorie" {...input}>
                <option value="" disabled>
                  catégories...
                </option>
                {Object.entries(optionCategorie).map(([value, label]) => (
                  <option key={value} value={value}>
                    {label}
                  </option>
                ))}
              </Select>
            </FormGroup>
          );
        }}
      </Field>
    </>
  );
}

export { Step };

const { spacings } = theme;
const Select = styled.select`
  max-width: 32rem;
`;

const FormGroup = styled.div`
  display: flex;
  flex-direction: column;
  align-items: flex-start;
  margin-bottom: ${spacings.medium};
`;
