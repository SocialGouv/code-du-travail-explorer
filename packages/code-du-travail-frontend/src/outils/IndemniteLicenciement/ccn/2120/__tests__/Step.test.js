import React from "react";
import { render } from "@testing-library/react";
import { Step } from "../Step";
import { Form } from "react-final-form";

describe("<Step />", () => {
  it("should render", () => {
    const onSubmit = jest.fn();
    const { container } = render(
      <Form onSubmit={onSubmit}>{({ form }) => <Step form={form} />}</Form>
    );
    expect(container).toMatchSnapshot();
  });
});
