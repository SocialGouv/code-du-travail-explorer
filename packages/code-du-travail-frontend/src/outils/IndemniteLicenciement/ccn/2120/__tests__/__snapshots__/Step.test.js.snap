// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Step /> should render 1`] = `
.c2 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  margin-right: 2em;
  padding: 0;
}

.c0 {
  margin: 0 0 1rem 0;
  color: #2f3b6c;
  font-weight: normal;
  font-size: 2.8rem;
  font-family: "Merriweather",serif;
  line-height: 1.25;
  margin-top: 3.2rem;
  margin-bottom: 2rem;
}

.c3 {
  max-width: 32rem;
}

.c1 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  -webkit-align-items: flex-start;
  -webkit-box-align: flex-start;
  -ms-flex-align: flex-start;
  align-items: flex-start;
  margin-bottom: 2rem;
}

<div>
  <h2
    class="c0"
  >
    Motif du licenciement
  </h2>
  <div
    class="c1"
  >
    <label
      class="c2"
      for="input-motif"
    >
      Préciser le motif de licenciement :
    </label>
    <select
      class="c3"
      id="input-motif"
      name="motif"
    >
      <option
        disabled=""
        value=""
      >
        motifs...
      </option>
      <option
        value="disciplinaire"
      >
        Motif disciplinaire
      </option>
      <option
        value="non-disciplinaire"
      >
        Motif non disciplinaire
      </option>
      <option
        value="eco"
      >
        Motif économique
      </option>
    </select>
  </div>
  <div
    class="c1"
  >
    <label
      class="c2"
      for="input-categorie"
    >
      Précisez la categorie professionnelle du salarié :
    </label>
    <select
      class="c3"
      id="input-categorie"
      name="categorie"
    >
      <option
        disabled=""
        value=""
      >
        catégories...
      </option>
      <option
        value="non-cadre"
      >
        Non cadre
      </option>
      <option
        value="cadre"
      >
        Cadre
      </option>
      <option
        value="ne-sais-pas"
      >
        Ne sait pas
      </option>
    </select>
  </div>
</div>
`;
